### ISSUE TITLE
[Meta issue] Ship 1.0

### ISSUE THREAD

**Issue by igorT** [emberjs/data#2545]
Target date: June 12th.
Num days left: 0
**Blocked for Tomhuda review**: https://github.com/emberjs/data/issues/3298 -reviewed

Current status:
@igorT coordinating, relationships refactor, refactor to use json api internally
@wecc json api Adapter, pagination, metadata
@tchak errors object
@bmac refactor of the find mechanism
@fivetanley ?
@pangratz working on the references RFC
@sly7-7 free?

Done:
- [x] implement the Snapshots API (@wecc)
- [x] move `FixtureAdapter` into a plugin (#2804) @bmac 
- [x] Refactor the internals to the new json api format @wecc 
- [x] Refactor the serializers to follow jsonapi (#3143) @wecc
- [x] improve attribute change triggering (#2998) @wecc
- [x] move model state out of the constructor (#2907)
- [x] audit the PRs
- [x] audit the issues

In progress:
- [ ] cleanup the errors object (@tchak has a branch, cc @bcardarella)
- [ ] deleted records get immediately removed from live arrays #1737 #2867
- [ ] unloaded records get immediately destroyed #2870
- [ ] JSONAPI adapter (#2905) @wecc
- [ ] cleanup metadata handling @wecc 
- [ ] refactor the find mechanism @bmac 
- [ ] audit the code for TODOs, other crap, etc.
- [ ] peek 
- [ ] implement `queryOne` (#2584)
- [ ] make async relationships the default #3220
- [ ] references @pangratz 
- [ ] relationship feeding @pangratz 

Descoped:
- [ ] Test out async es7 babbel await support/workflow for relationships - blocked on jshint support
- [ ] clean up polymorphism #2991 #2775
- [ ] cleanup the state manager (state management is shared by the state manager and record, would be better in 1 place )
- [ ] make model fragments work with 1.0 @slindberg 
- [ ] support for streaming deletes and relationship changes 
- [ ] finish relationship refactor so that we support proper rollback/dirtyness of relationships (@igorT) (#1367)
- [ ] use the new metadata for nice pagination @wecc 
- [ ] JSONApi filtering

**Comment by wecc** [comment]
Use #602 for "internal properties conflict with user defined properties"?

**Comment by wycats** [comment]
@wecc @igorT @tomdale @bmac I would like to break out these checkboxes into separate issues, and then triage them to prune things that can be added compatibly post 1.0.

In Rust, there is a tag called `P-BackCompat`, which basically means: "1.0 priority because it can't be added compatibly later". We can do something similar.

**Comment by thomassnielsen** [comment]
Some of the checkboxes could need some clarification. As an example, `unloaded records get immediately destroyed ` can be interpreted both as something to be implemented and as a bug to be fixed.

**Comment by knownasilya** [comment]
Is ember-data at a stand still? Is there an ETA? Just wondering, since Ember is approaching v2, and ED is still not even at 1.0..

**Comment by locks** [comment]
A new beta came out recently (last week, I think?) and jsonapi is nearly
finalised so is a matter of time

**Comment by locks** [comment]
Oh and it was announced in Emberconf that ED 1.0 is coming out June 20th
with Ember 2.0

**Comment by igorT** [comment]
Closing this in favor of 
https://github.com/emberjs/data/issues/3320
https://github.com/emberjs/data/issues/3321
Almost there
